### PR TITLE
vcftools-consensus from vcftools

### PR DESCRIPTION
**Pull request by pvanheus**
This provides the vcftools-consensus tool that merges a FASTA and VCF file to generate a new FASTA sequence derived from applying the variants in the VCF file to the FASTA reference sequence. 

I am contributing it here because of the vcftools collection in this tools-devteam repo but could equally add it to tools-iuc since there is nothing, per se, shared with the collection (macros are not used, for example).

### CODE REVIEW

**Comment by bgruening** [comment]
@pvanheus can you please ass a .shed.yml file for this tool?

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<tool id="vcftools_consensus" name="Consensus" version="0.1.11">
+    <description>Apply VCF variants to a fasta file to create consensus sequence</description>
+
+    <requirements>
+        <requirement type="package">samtools</requirement>
+        <requirement type="package">htslib</requirement>
```
Comment (bgruening) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Can you please add versions.

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    <stdio>
+        <!--<regex match=".*" source="both" level="log" description="tool progress"/>-->
+        <exit_code range="1:" level="fatal" />
+    </stdio>
+
+    <command>
+    <![CDATA[
+        ln -s "${variants}" variants.vcf &&
+        vcf-sort variants.vcf >variants.sorted.vcf &&
```
Comment (bgruening) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Can we pipe all three commands?

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml (hunk 1 above)
Comment (pvanheus) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Are version numbers required? From samtools faidx is used, from htslib bgzip and tabix are used. As far as I know any versions of these tools will work.

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml (hunk 2 above)
Comment (pvanheus) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Changed to pipe vcf-sort to bgzip, can't pipe to tabix as far as I can see.

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml (hunk 1 above)
Comment (mvdbeek) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: But that might not be true anymore in a year or two. 
You can just add the latest and greatest.

**Comment by bgruening** [comment]
:+1:

File: tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml (hunk 1 above)
Comment (bgruening) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Strictly it is not required, but it is good for reproducibility as Marius mentioned.

Comment (nsoranzo) [inline tool_collections/vcftools/vcftools_consensus/vcftools_consensus.xml]: Should we add this to https://github.com/galaxy-iuc/standards/blob/master/docs/best_practices/tool_xml.rst ?

**Comment by nsoranzo** [comment]
@martenson It seems the automatic upload to TTS and MTS failed (see https://travis-ci.org/galaxyproject/tools-devteam/jobs/173334026 ), can you push this manually to both, please?

**Comment by martenson** [comment]
@nsoranzo both TSs seem to be up to date with this tool

**Comment by nsoranzo** [comment]
@martenson It has been auto-uploaded when #417 was merged, sorry I forgot to inform you.

**Pull request merged**